### ISSUE TITLE
Plugin panics collecting metrics from cirros vm

### ISSUE THREAD

**Issue by jcooklin** [intelsdi-x/snap-plugin-collector-libvirt#32]
**Snap version** (use `snapctl -v`):
latest (snapd version master-8fd4664)

**Environment**:

```
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:    14.04
Codename:   trusty
```

```
QEMU emulator version 2.0.0 (Debian 2.0.0+dfsg-2ubuntu1.19), Copyright (c) 2003-2008 Fabrice Bellard
libvirtd (libvirt) 1.2.2
```

```
▶ virsh list
 Id    Name                           State
----------------------------------------------------
 2     cirros                         running
```

snap task:

```

---
  version: 1
  schedule:
    type: "simple"
    interval: "1s"
  workflow:
    collect:
      metrics:
        /libvirt/cirros/cpu/*: {}
        /libvirt/cirros/mem/*: {}
        /libvirt/cirros/disk/*: {}
      publish:
        -
          plugin_name: "file"
          config:
            file: "/tmp/snap-libvirt.out"
```

**What happened**:
 **Panic**.  I can collect if I remove the metrics related to disks.  This VM doesn't have a disk.  Ultimately I want to collect `/libvirt/*`.  

![img](https://www.evernote.com/l/AA1K-u-AoTpOHqqL10wB082J1AN5djOn_EkB/image.png)

**What you expected to happen**:
I expected it to work and return zero values for metrics under disk.

**Steps to reproduce it** (as minimally and precisely as possible):
1. create cirros image
2. start task above 

**Anything else do we need to know** (e.g. issue happens only occasionally):

**Comment by sandlbn** [comment]
@jcooklin 

Plugin doesn't support, dynamic metrics for disk and network devices. I will be working on that next week along with the PR for extended support for Openstack nova metadata. This bug is related to the issue #28 . Now you can get metrics using namespace /libvirt/cirros/disk/{device_name}/rdbytes .

**Comment by dgarth** [comment]
I am not sure if this is a new issue or belongs here. While you can now use wildcards in the metrics the plugin still fails for me after starting a task which collects disk data.

**Environment**
CentOS 7 (Kernel 3.10.0-327.36.3.el7.x86_64)
Snap v0.16.1-beta
OpenStack Liberty (via Devstack)
Libvirt 1.2.17-13.el7_2.5
qemu-kvm 10:2.3.0-31

These metrics are collected:
```
/libvirt/*/disk/*/wrreq
/libvirt/*/disk/*/rdreq
/libvirt/*/disk/*/wrbytes
/libvirt/*/disk/*/rdbytes
```


[Log snippet](http://pastebin.com/ie5yVEmL)

**Comment by sandlbn** [comment]
Fixed in #36
